src/features/navbar/components/BitwiseNavbar.tsx

```diff
@@ -0,0 +1,191 @@
+    }
+
+    return light.nav.horizontal.backgroundColor;
+  }
+
+  if (isMobilePerspective) {
+    return dark.nav.vertical.backgroundColor;
+  }
+
+  return dark.nav.horizontal.backgroundColor;
+};
+
+export const BitwiseNavbar: FC<Props> = ({ closeVerticalNav }) => {
+  const { user } = useAuth();
+  const navLinks = useNavLinks();
+  const { logout, isLoading } = useLogout();
+  const { i18n } = useTranslation();
+  const { innerWidth: width } = window;
+  const [isMobilePerspective, setIsMobilePerspective] = useState(width <= 767);
+  const { theme } = useTheme();
+  const navbarMarginTop = environment.environment === EnvironmentConfiguration.Staging ? '56px' : '0px';
+  const offcanvasMarginTop =
+    environment.environment === EnvironmentConfiguration.Staging && isMobilePerspective ? '56px' : '0px';
+  const bgColor = getNavBackgroundColor(isMobilePerspective, theme);
+
+  const [showModal, hideModal] = useModal(
+    ({ in: open, onExited }) => {
+      return (
+        <Modal show={open} onHide={hideModal} onExited={onExited}>
+          <Modal.Header closeButton>
+            <Modal.Title>Sign Out</Modal.Title>
+          </Modal.Header>
+          <Modal.Body>Are you sure you want to sign out?</Modal.Body>
+          <Modal.Footer>
+            <Button variant='link' onClick={hideModal}>
+              Cancel
+            </Button>
+            <LoadingButton className='action-shadow' loading={isLoading} variant='primary' onClick={() => logout()}>
+              <FontAwesomeIcon icon='sign-out-alt' /> Sign Out
+            </LoadingButton>
+          </Modal.Footer>
+        </Modal>
+      );
+    },
+    [isLoading],
+  );
+
+  const changeLanguage = (ln: string) => {
+    localStorage.setItem('language', ln);
+    i18n.changeLanguage(ln);
+  };
+
+  const __languageOptions = languages.map(language => {
+    return { label: language.label, value: language.shortcode };
+  });
+
+  const defaultLanguageOption = __languageOptions.find(language => language.value === i18n.languages[0]);
+
+  const handleResize = () => {
+    const newWidth = window.innerWidth;
+    if (newWidth <= 767 && !isMobilePerspective) {
+      setIsMobilePerspective(true);
+    } else if (newWidth > 767 && isMobilePerspective) {
+      setIsMobilePerspective(false);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('resize', handleResize);
+  });
+
+  return (
+    <Navbar
+      expand='md'
+      className='position-fixed mb-3 w-100'
+      style={{ backgroundColor: bgColor, zIndex: 1030, marginTop: navbarMarginTop }}
+    >
+      <Container fluid>
+        <Navbar.Brand href='#'>
+          <Logo />
+        </Navbar.Brand>
+        <Navbar.Toggle aria-controls='offcanvasNavbar-expand-md' />
+        <Navbar.Offcanvas
+          id='offcanvasNavbar-expand-md'
+          aria-labelledby='offcanvasNavbarLabel-expand-md'
+          placement='end'
+          style={{ backgroundColor: bgColor, marginTop: offcanvasMarginTop }}
+        >
```

Comment:
Excellent use of offcanvas here! I tested this out locally and the offcanvas response stuff works great!